--- shout.h.orig	2010-05-01 00:19:21.000000000 +0200
+++ shout.h	2010-05-01 00:59:40.000000000 +0200
@@ -8,7 +8,7 @@
 #define PASSWORD "letmein"
 
 /* And the name */
-#define NAME "Icecast Radio"
+#define NAME "Freetz Radio"
 
 /* And the genre */
 #define GENRE "misc"
@@ -27,7 +27,7 @@
    shout.log
 */
 #ifndef _WIN32
-#define DEFAULT_BASE_DIR "/tmp"
+#define DEFAULT_BASE_DIR "/var/tmp/flash/mod"
 #endif
 
 #ifndef _WIN32
@@ -38,7 +38,7 @@
 
 /* Run system (DJPROGRAM) before every song
    Please note that this is terribly inefficient */
-#define DJPROGRAM "/usr/local/bin/im_the_dj -with -lots -of -cool -options"
+#define DJPROGRAM "/var/tmp/flash/mod/im_the_dj -with -lots -of -cool -options"
 
 /* This is the default config file. 
    Note that this _always_ gets parsed, even if you supply another
@@ -69,10 +69,10 @@
 #define READ 101
 
 /* This should be set to something optimal on your system */
-#define DEFAULT_BUFFER_SIZE 4096
+#define DEFAULT_BUFFER_SIZE 1024
 
 /* This is just for scream() */
-#define BUFSIZE 1024
+#define BUFSIZE 512
 
 #ifdef __GNUC__
 #define HAVE_LONG_LONG
