--- playaudio.c.orig	2011-05-22 09:41:50.000000000 +0200
+++ playaudio.c	2011-05-22 09:44:09.000000000 +0200
@@ -166,7 +166,7 @@
  * Main loop of child process which keeps an MPEG player running. */
 static void mpeg_player_manager(void) {
     extern sig_atomic_t foad; /* in driftnet.c */
-    struct sigaction sa = {0};
+    struct sigaction sa = {{0}};
     pid_t mpeg_pid;
 
     sa.sa_handler = SIG_DFL;
