--- sshow.c.orig	2011-04-30 09:24:54.000000000 +0200
+++ sshow.c	2011-04-30 09:19:07.000000000 +0200
@@ -15,6 +15,7 @@
 
 #include <sys/types.h>
 #include <sys/times.h>
+#include <time.h>
 
 #include <netinet/in_systm.h>
 #include <netinet/in.h>
@@ -41,6 +42,10 @@
 
 #define HISTORY_SIZE			16
 
+#ifndef CLK_TCK
+# define CLK_TCK	CLOCKS_PER_SEC
+#endif
+
 typedef struct {
 	u_int min, max;
 } range;
