--- config.h.orig	2011-07-15 23:04:01.000000000 +0200
+++ config.h	2011-07-16 12:46:12.000000000 +0200
@@ -2,17 +2,38 @@
  * Configuration file for psyBNC, created by menuconf
  */
 
+#undef IPV6
+#undef TRANSLATE
+#undef TRAFFICLOG
+#undef LINKAGE
+#undef BLOWFISH
+
+/*
+#undef CRYPT
+#undef BLOWFISH
+#undef IDEA
+#undef HAVE_SSL
+#undef PARTYCHANNEL
+#undef DCCFILES
+#undef DCCCHAT
+#define PROXYS
+*/
+
+#define NETWORK
+
 /* Encryption */
 
 #define CRYPT
+#define HAVE_SSL
 
 /* Encryption Type*/
 
-#define BLOWFISH
+/*#define BLOWFISH*/
+#define IDEA
 
 /* Allow Translation */
 
-#define TRANSLATE
+/*#define TRANSLATE*/
 
 /* Allow internal network */
 
@@ -20,11 +41,11 @@
 
 /* Allow traffic logging */
 
-#define TRAFFICLOG
+/*#define TRAFFICLOG*/
 
 /* Allow linkage of bouncers */
 
-#define LINKAGE
+/*#define LINKAGE*/
 
 /* Allow the dcc File-Functions */
 
@@ -40,11 +61,11 @@
 
 /* Number of max. Users */
 
-#define MAXUSER 50
+#define MAXUSER 5
 
 /* Number of max. Connections per User */
 
-#define MAXCONN 99
+#define MAXCONN 5
 
 /* Use blocking DNS */
 
@@ -56,9 +77,26 @@
 
 /* The logging level */
 
-#define LOGLEVEL 0
+#define LOGLEVEL 2
 
 /* SSL-Security */
 
-#define SSLSEC 2
+#define SSLPATH "/tmp/"
+#define SSLSEC 0
 
+/* http://www.exodus.ro/psybnc.html */
+/* http://wiki.globalshellz.org/gsh/PsyBNC_HOWTO */
+/* http://www.lunarshells.com/install_psybnc.php#config */
+/*
+#define IDEA
+#define ANONYMOUS
+#define PROXYS
+#define DYNAMIC 
+#define PARTYCHANNEL
+#define CTCPVERSION "xxx"
+#define OIDENTD
+#define SCRIPTING
+#define SINGLEUSER
+#define NETWORK
+#define SSLPATH "/path/to/ssl"
+*/
