--- Makefile.orig	2011-09-07 15:32:01.000000000 +0200
+++ Makefile	2011-09-07 15:34:25.000000000 +0200
@@ -1,25 +1,25 @@
 # Configuration
 
 VERSION="v1.9"
-USELIBWRAP=	# Use libwrap?
-PREFIX=/usr/local
+#USELIBWRAP=	# Use libwrap?
+PREFIX=/usr
 
 MAN=sslh.8.gz	# man page name
 
 # End of configuration -- the rest should take care of
 # itself
 
-CC = gcc
-CFLAGS=-Wall -g
+CC=
+CFLAGS=
 
 #LIBS=-lnet
-LIBS=
+LIBS=-lwrap
 OBJS=common.o
 
-ifneq ($(strip $(USELIBWRAP)),)
-	LIBS:=$(LIBS) -lwrap
-	CFLAGS:=$(CFLAGS) -DLIBWRAP
-endif
+#ifneq ($(strip $(USELIBWRAP)),)
+#	LIBS:=$(LIBS) -lwrap
+#	CFLAGS:=$(CFLAGS) -DLIBWRAP
+#endif
 
 all: sslh $(MAN)
 
@@ -31,11 +31,11 @@
 
 sslh-fork: $(OBJS) sslh-fork.o Makefile common.h
 	$(CC) $(CFLAGS) -D'VERSION=$(VERSION)' -o sslh-fork sslh-fork.o $(OBJS) $(LIBS)
-	strip sslh-fork
+#	strip sslh-fork
 
 sslh-select: $(OBJS) sslh-select.o Makefile common.h 
 	$(CC) $(CFLAGS) -D'VERSION=$(VERSION)' -o sslh-select sslh-select.o $(OBJS) $(LIBS)
-	strip sslh-select
+#	strip sslh-select
 
 
 $(MAN): sslh.pod Makefile
