--- Makefile.orig	2012-04-22 13:21:28.019641097 +0200
+++ Makefile	2012-04-22 13:22:46.411644537 +0200
@@ -1,9 +1,9 @@
 # Configuration
 
 VERSION="v1.11"
-USELIBWRAP=	# Use libwrap?
+#USELIBWRAP=	# Use libwrap?
 COV_TEST= 	# Perform test coverage?
-PREFIX=/usr/local
+PREFIX=/usr
 
 MAN=sslh.8.gz	# man page name
 
@@ -14,17 +14,17 @@
     CFLAGS_COV=-fprofile-arcs -ftest-coverage
 endif
 
-CC = gcc
-CFLAGS=-Wall -g $(CFLAGS_COV)
+CC =
+CFLAGS=
 
 #LIBS=-lnet
-LIBS=
+LIBS=-lwrap
 OBJS=common.o sslh-main.o
 
-ifneq ($(strip $(USELIBWRAP)),)
-	LIBS:=$(LIBS) -lwrap
-	CFLAGS:=$(CFLAGS) -DLIBWRAP
-endif
+#ifneq ($(strip $(USELIBWRAP)),)
+#	LIBS:=$(LIBS) -lwrap
+#	CFLAGS:=$(CFLAGS) -DLIBWRAP
+#endif
 
 all: sslh $(MAN) echosrv
 
