--- src/iplog.h	2011-12-31 17:47:19.000000000 +0100
+++ src/iplog.h	2011-12-31 17:52:51.876452646 +0100
@@ -183,7 +183,6 @@
 int udp_parser(const struct ip *ip);
 int icmp_parser(const struct ip *ip);
 
-u_char *inet_ntoa_r(const struct in_addr *in, u_char *buf, size_t len);
 u_char *host_lookup(const struct in_addr *in, bool resolv, u_char *buf, size_t len);
 u_char *_host_lookup(const struct in_addr *in, u_char *buf, size_t len);
 
--- src/iplog_util.c	2001-01-01 17:02:14.000000000 +0100
+++ src/iplog_util.c	2011-12-31 17:52:46.648517740 +0100
@@ -330,20 +330,6 @@
 #endif
 
 /*
-** Thread-safe version of inet_ntoa(3).
-*/
-
-u_char *inet_ntoa_r(const struct in_addr *in, u_char *buf, size_t len) {
-	ipaddr_t addr = ntohl(in->s_addr);
-
-	snprintf(buf, len, "%u.%u.%u.%u",
-			((addr >> 24) & 0xff), ((addr >> 16) & 0xff),
-			((addr >> 8) & 0xff), (addr & 0xff));
-
-	return (buf);
-}
-
-/*
 ** Resolve a protocol number to its name (eg. 6 -> "TCP")
 ** returns "proto %u" on failure.
 */
