--- src/checksum.c.orig	2011-04-29 13:07:24.000000000 +0200
+++ src/checksum.c	2011-04-29 20:15:21.000000000 +0200
@@ -5,6 +5,7 @@
 #include <netinet/ip.h>
 #include <netinet/tcp.h>
 #include <netinet/udp.h>
+
 #include "nids.h"
 
 static struct nids_chksum_ctl * nchk;
@@ -25,7 +26,7 @@
 	return 0;
 }
  
-#if ( __i386__ || __i386 )
+#if ( 0 )
 // all asm procedures are copied from Linux 2.0.36 and 2.2.10 kernels
 
 /*
