--- Makefile.orig	2011-02-05 09:47:21.000000000 +0100
+++ Makefile		2011-02-05 10:02:44.000000000 +0100
@@ -1,21 +1,6 @@
-TARGET = bost
-SRCS = bost.c
+CFLAGS=
+CC=
+LDFLAGS=
 
-CC = gcc
-LD = gcc
-
-CFLAGS = -std=gnu99 -Wall -pedantic -D_GNU_SOURCE -g #-DDEBUG
-LDFLAGS = -g -lm
-
-OBJS = $(SRCS:.c=.o)
-
-all:$(TARGET)
-
-$(TARGET):$(OBJS)
-	$(LD) -o $@ $^ $(LDFLAGS)
-
-%.o:%.c
-	$(CC) -c $< $(CFLAGS)
-
-clean:
-	rm -f $(TARGET) *.o *~ *.bak
+bost: bost.c
+		${CC} ${CFLAGS} -o bost bost.c ${LDFLAGS}
