--- calltable.h.orig	2006-08-13 12:56:46.000000000 +0200
+++ calltable.h	2011-04-24 12:29:17.000000000 +0200
@@ -30,14 +30,18 @@
 
 struct calltable_element {
 	unsigned char is_used;
+	unsigned char had_bye;
+	unsigned char had_t38;
 	char call_id[32];
 	unsigned long call_id_len ;
 	in_addr_t ip[calltable_max_ip_per_call];
-	unsigned short port[calltable_max_ip_per_call];
+        uint16_t port[calltable_max_ip_per_call];
+        uint32_t ssrc[calltable_max_ip_per_call];
 	int ip_n;
 	time_t last_packet_time;
 	pcap_dumper_t *f_pcap;
 	FILE *f;
+	char fn_pcap[128];
 };
 
 class calltable
@@ -58,8 +62,15 @@
 	int find_ip_port(
 	    in_addr_t addr,
 	    unsigned short port);
+        int find_ip_port_ssrc(
+            in_addr_t addr,
+            unsigned short port,
+            uint32_t ssrc,
+            int *idx_leg,
+            int *idx_rtp);
 	int do_cleanup( time_t currtime );
 	calltable_element *table;
+	bool erase_non_t38;
     private:
 	unsigned long table_size;
 	time_t global_last_packet_time;
