--- pcapsipdump.h.orig	2007-05-11 20:40:49.000000000 +0200
+++ pcapsipdump.h	2011-04-24 12:29:17.000000000 +0200
@@ -26,7 +26,7 @@
     This would be appreciated, but not required.
 */
 
-#define PCAPSIPDUMP_VERSION "0.1.4"
+#define PCAPSIPDUMP_VERSION "0.1.4-trunk"
 
 struct iphdr {
 #if defined(__LITTLE_ENDIAN)
@@ -35,7 +35,7 @@
 #elif defined (__BIG_ENDIAN)
 	uint8_t	version:4,
   		ihl:4;
-#elif
+#else
 #error Endian not defined
 #endif
 	uint8_t	tos;
@@ -57,3 +57,14 @@
 	uint16_t	len;
 	uint16_t	check;
 };
+
+#ifndef ETH_ALEN
+#define ETH_ALEN 6
+#endif
+
+struct ether_header
+{
+  uint8_t  ether_dhost[ETH_ALEN];	/* destination eth addr	*/
+  uint8_t  ether_shost[ETH_ALEN];	/* source ether addr	*/
+  uint16_t ether_type;		        /* packet type ID field	*/
+} __attribute__ ((__packed__));
