--- Makefile.orig	2011-04-15 20:43:49.000000000 +0200
+++ Makefile	2011-04-15 20:46:18.000000000 +0200
@@ -5,8 +5,8 @@
 OUT := redsocks
 
 LIBS := -levent
-CFLAGS += -g -O2
-override CFLAGS += -std=gnu99 -Wall
+CFLAGS +=
+CC =
 
 all: $(OUT)
 
@@ -32,7 +32,7 @@
 base.c: $(CONF)
 
 $(DEPS): $(SRCS)
-	gcc -MM $(SRCS) 2>/dev/null >$(DEPS) || \
+	$(CC) -MM $(SRCS) 2>/dev/null >$(DEPS) || \
 	( \
 		for I in $(wildcard *.h); do \
 			export $${I//[-.]/_}_DEPS="`sed '/^\#[ \t]*include \?"\(.*\)".*/!d;s//\1/' $$I`"; \
