--- Makefile.orig	2007-02-09 12:33:48.000000000 +0100
+++ Makefile	2011-04-24 12:29:17.000000000 +0200
@@ -1,17 +1,19 @@
+LIBS= -lpcap -lstdc++
+
 all: pcapsipdump
 
 pcapsipdump: pcapsipdump.cpp calltable.cpp calltable.h
-	$(CC) $(CPPFLAGS) $(LDFLAGS) pcapsipdump.cpp calltable.cpp -o pcapsipdump -lpcap -lstdc++
+	$(CC) $(CPPFLAGS) $(LDFLAGS) $(LIBS) pcapsipdump.cpp calltable.cpp -o pcapsipdump
 
 pcapsipdump-debug: pcapsipdump.cpp calltable.cpp calltable.h
-	$(CC) $(CPPFLAGS) $(LDFLAGS) -ggdb pcapsipdump.cpp calltable.cpp -o pcapsipdump-debug -lpcap -lstdc++
+	$(CC) $(CPPFLAGS) $(LDFLAGS) $(LIBS) -ggdb pcapsipdump.cpp calltable.cpp -o pcapsipdump-debug
 
 clean:
 	rm -f pcapsipdump
 
 install:
 	install pcapsipdump ${DESTDIR}/usr/sbin/pcapsipdump
-	install redhat/pcapsipdump.init ${DESTDIR}/etc/rc.d/init.d/pcapsipdump
+	install redhat/pcapsipdump.init ${DESTDIR}/etc/init.d/pcapsipdump
 	install redhat/pcapsipdump.sysconfig ${DESTDIR}/etc/sysconfig/pcapsipdump
 	mkdir -p ${DESTDIR}/var/spool/pcapsipdump
 	chmod 0700 ${DESTDIR}/var/spool/pcapsipdump
