--- src/p_socket.c.orig	2007-03-20 13:41:44.000000000 +0100
+++ src/p_socket.c	2010-01-06 23:42:28.000000000 +0100
@@ -242,6 +242,7 @@
   char *ho;
   int listensocket;
   int rc;
+  char in6addr_any;
   const char *pt;
   vsl[0]=0;
   if(host==NULL) return 0;
--- src/p_socket.c.orig	2007-03-20 13:41:44.000000000 +0100
+++ src/p_socket.c	2010-01-06 23:42:28.000000000 +0100
@@ -242,6 +242,7 @@
   char *ho;
   int listensocket;
   int rc;
+  char in6addr_any;
   const char *pt;
   vsl[0]=0;
   if(host==NULL) return 0;
