--- process/sysdep_LINUX.c.orig	2011-04-09 01:49:23.000000000 +0200
+++ process/sysdep_LINUX.c	2011-04-09 01:48:43.000000000 +0200
@@ -88,6 +88,8 @@
 # define HZ sysconf(_SC_CLK_TCK)
 #endif
 
+# define GLOB_ONLYDIR    (1 << 13)/* Match only directories.  */
+
 #include "monitor.h"
 #include "process.h"
 #include "process_sysdep.h"
@@ -144,6 +146,45 @@
 
   return time(NULL) - (time_t)up;
 }
+
+
+/* uclibc and dietlibc do not have this junk -ReneR */
+#if defined (__UCLIBC__) || defined (__dietlibc__)
+static int getloadavg (double loadavg[], int nelem)
+{
+  int fd;
+
+  fd = open ("/proc/loadavg", O_RDONLY);
+  if (fd < 0)
+    return -1;
+  else
+    {
+      char buf[65], *p;
+      ssize_t nread;
+      int i;
+
+      nread = read (fd, buf, sizeof buf - 1);
+      close (fd);
+      if (nread <= 0)
+	return -1;
+      buf[nread - 1] = '\0';
+
+      if (nelem > 3)
+	nelem = 3;
+      p = buf;
+      for (i = 0; i < nelem; ++i)
+	{
+	  char *endp;
+	  loadavg[i] = strtod (p, &endp);
+	  if (endp == p)
+	    return -1;
+	  p = endp;
+	}
+
+      return i;
+    }
+}
+#endif
   
 
 /* ------------------------------------------------------------------ Public */
