--- ipv4.h.orig	2002-02-08 00:52:38.000000000 +0100
+++ ipv4.h	2011-05-20 23:35:29.000000000 +0200
@@ -63,7 +63,7 @@
  */
 sendip_option ip_opts[] = {
 	{"s",1,"Source IP address (see README)","127.0.0.1"},
-	{"d",1,"Desitnation IP address","Correct"},
+	{"d",1,"Destination IP address","Correct"},
 	{"h",1,"IP header length (see README)","Correct"},
 	{"v",1,"IP version (you almost definately don't want to change this)","4"},
 	{"y",1,"IP type of service","0"},
